src/main/java/com/mraof/minestuck/inventory/musicplayer/CapabilityProviderMusicPlayer.java

```diff
@@ -0,0 +1,58 @@
+package com.mraof.minestuck.inventory.musicplayer;
+
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class CapabilityProviderMusicPlayer implements ICapabilityProvider, INBTSerializable<CompoundTag>
+{
+	private final LazyOptional<IItemHandler> lazyInitialisionSupplierItemHandler = LazyOptional.of(this::getCachedInventory);
+	private final LazyOptional<IMusicPlaying> lazyInitialisionSupplierMusicPlaying = LazyOptional.of(this::getCachedMusicPlaying);
+	private ItemStackHandlerMusicPlayer itemStackHandlerMusicPlayer;
+	private MusicPlaying musicPlaying;
+	
+	@NotNull
+	@Override
+	public <T> LazyOptional<T> getCapability(@NotNull Capability<T> cap, @Nullable Direction side)
+	{
+		if(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY == cap)
+			return (LazyOptional<T>) (lazyInitialisionSupplierItemHandler);
```

Comment:
A slightly cleaner way of doing the cast is to call `lazyOptional.cast()` instead of doing the cast directly.

Reply:
oh ok !